workflow/controller/operator.go

```diff
@@ -1922,10 +1879,70 @@ func (woc *wfOperationCtx) executeTemplate(ctx context.Context, nodeName string,
 					return nil, err
 				}
 			}
+			// Set this value to check that this node is using synchronization, and acquire the lock.
+			unlockedNode = true
 		}
 
 		woc.updated = woc.updated || wfUpdated
 	}
+
+	// Check memoization cache if the node is about to be created, or was created in the past but is only now allowed to run due to acquiring a lock
+	if processedTmpl.Memoize != nil {
+		if node == nil || unlockedNode {
+			memoizationCache := woc.controller.cacheFactory.GetCache(controllercache.ConfigMapCache, processedTmpl.Memoize.Cache.ConfigMap.Name)
+			if memoizationCache == nil {
+				woc.log.WithFields(log.Fields{"cacheName": processedTmpl.Memoize.Cache.ConfigMap.Name}).WithError(err)
+				return woc.initializeNodeOrMarkError(node, nodeName, templateScope, orgTmpl, opts.boundaryID, err), err
+			}
```

Comment:
I noticed this line in the original code: `err := fmt.Errorf("cache could not be found or created")`
Was that removed on purpose?

Reply:
Good catch @juliev0. It needs to go back in (or something similar), otherwise `err` used on new line 1894 and 1895 will be from somewhere else (1802 probably).

Reply:
oops, another mistake. I'll add that line as original code does.